File: CodeForces/546A-Soldier and Bananas/546a.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
  long long int price, dolars, bananas;
  int totalPrice = 0;
  cin >> price >> dolars >> bananas;

  for(int i = 1; i <= bananas; ++i){
    totalPrice += price * i;
  }

  int sol = totalPrice - dolars;
  
  if(sol <= 0) cout << 0 << endl;
  else cout << sol << endl;
  
  return 0;
}

File: CodeForces/546A-Soldier and Bananas/546a.java
import java.util.*;

class Main{

    public static void main(String[] args){
	Scanner sc = new Scanner(System.in);
	int price = sc.nextInt();
	int dolars = sc.nextInt();
	int bananas = sc.nextInt();
	int total = 0;
	int i = 1;
	
	while(i <= bananas){
	    total += price * i;
	    ++i;
	}
	
	if(total-dolars <= 0) System.out.println(0);
	else System.out.println(total-dolars);
        
	sc.close();
    }
}
